tests/functional/test_httpchecksum.py

```diff
@@ -549,3 +598,96 @@ def test_unsuccessful_streaming_response_checksum_validation(
             expected_checksum,
         )
         assert str(expected_error.value) == error_msg
+
+
+def _checksum_user_agent_feature_id_cases():
+    request_payload = "Hello world"
+    cases = [
+        (None, None, None, request_payload, ["U", "b", "Z"]),
+        (
+            "CRC32",
+            "when_required",
+            "when_required",
+            request_payload,
+            ["U", "a", "c"],
+        ),
+        (
+            "SHA1",
+            "when_supported",
+            "when_supported",
+            request_payload,
+            ["X", "Z", "b"],
+        ),
+        (
+            "SHA256",
+            "when_supported",
+            "when_required",
+            request_payload,
+            ["Y", "Z", "c"],
+        ),
+        (
+            "SHA1",
+            "when_required",
+            "when_supported",
+            request_payload,
+            ["X", "a", "b"],
+        ),
+        ("SHA256", None, "when_supported", request_payload, ["Y", "Z", "b"]),
+        ("SHA1", "when_required", None, request_payload, ["X", "a", "b"]),
```

Comment:
The first case already tests the default values so I don't think these are needed